sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuColumnVector.java

```diff
@@ -202,7 +220,21 @@ static final DataType getSparkType(DType type) {
     DType type = access.getDataType();
     if (type == DType.LIST) {
       try (ColumnViewAccess<T> child = access.getChildColumnViewAccess(0)) {
-        return new ArrayType(getSparkTypeFrom(child), true);
+        DataType ret;
```

Comment:
Ah I see. good idea, I will work on that.

Reply:
@revans2  The latest changes try to address this specific comment 
> where we have to pass in the Spark DataType when changing to a ColumnarBatch instead of inferring it and getting it wrong. 

And also adds a check I mentioned earlier today in :
> I totally understand and get that it is terrible hack. I was going to address it in this PR with something like this (which seems to work for my test query but I would love to get your thoughts on it):
```
  override def coalesceAfter: Boolean = {
   !child.expressions.exists( expr =>  expr.dataType.isInstanceOf[MapType])
  }
```
> If this is not sufficient and needs more work on the overrides and coalesce batch side then I would love to have your PR be separate and I can help review and get it in before this does. Thank you so much.
>
I wanted your opinion on the changes to adding a `from()` with spark datatypes and whether we can simply it further and reduce the footprint. For starters I just pass the column datatypes as a list but happy to rework that and make the api be different. Just wanted to ask if I am headed the wrong or the right way here.